Code/CoreData/RKManagedObjectStore.m

```diff
@@ -399,11 +399,17 @@ - (BOOL)resetPersistentStores:(NSError **)error
                 }
             }
 
-            // Add a new store with the same options
-            NSPersistentStore *newStore = [self.persistentStoreCoordinator addPersistentStoreWithType:persistentStore.type
-                                                                                        configuration:persistentStore.configurationName
-                                                                                                  URL:persistentStore.URL
-                                                                                              options:persistentStore.options error:&localError];
+            NSString *seedPath = [persistentStore.options valueForKey:RKSQLitePersistentStoreSeedDatabasePathOption];
+            // Add a new store with the same options, except RKSQLitePersistentStoreSeedDatabasePathOption option
+            // that is not expected in this method.
+            NSMutableDictionary *mutableOptions = [persistentStore.options mutableCopy];
+            [mutableOptions removeObjectForKey:RKSQLitePersistentStoreSeedDatabasePathOption];
+            mutableOptions = [[mutableOptions allKeys] count] > 0 ? mutableOptions : nil;
```

Comment:
Ok, thank you, I will fix it tomorrow and add test coverage.

 

Sent from my iPhone

> On 18 Feb 2015, at 00:18, carllindberg <notifications@github.com> wrote:
> 
> In Code/CoreData/RKManagedObjectStore.m:
> 
> > @@ -399,11 +399,17 @@ - (BOOL)resetPersistentStores:(NSError **)error
> >                  }
> >              }
> >  
> > -            // Add a new store with the same options
> > -            NSPersistentStore *newStore = [self.persistentStoreCoordinator addPersistentStoreWithType:persistentStore.type
> > -                                                                                        configuration:persistentStore.configurationName
> > -                                                                                                  URL:persistentStore.URL
> > -                                                                                              options:persistentStore.options error:&localError];
> > +            NSString *seedPath = [persistentStore.options valueForKey:RKSQLitePersistentStoreSeedDatabasePathOption];
> > +            // Add a new store with the same options, except RKSQLitePersistentStoreSeedDatabasePathOption option
> > +            // that is not expected in this method.
> > +            NSMutableDictionary *mutableOptions = [persistentStore.options mutableCopy];
> > +            [mutableOptions removeObjectForKey:RKSQLitePersistentStoreSeedDatabasePathOption];
> > +            mutableOptions = [[mutableOptions allKeys] count] > 0 ? mutableOptions : nil;
> No need to call -allKeys there... dictionary has a -count method
> 
> —
> Reply to this email directly or view it on GitHub.
>